File: puzzles/ruby/temperatures.rb
n = gets.to_i # the number of temperatures to analyse
min = 5526
inputs = gets.split.map(&:to_i)
for t in inputs
  # t: a temperature expressed as an integer ranging from -273 to 5526
  if t.abs < min.abs || t.abs == min.abs && t > min
    min = t
  end
end

puts n == 0 ? 0 : min

File: puzzles/go/temperatures.go
package main

import "fmt"
import "os"
import "bufio"
import "strings"
import "strconv"
import "math"

func main() {
  scanner := bufio.NewScanner(os.Stdin)
  scanner.Buffer(make([]byte, 1000000), 1000000)

  // n: the number of temperatures to analyse
  var n int
  scanner.Scan()
  fmt.Sscan(scanner.Text(), &n)
  var minT = 5526.0

  scanner.Scan()
  inputs := strings.Split(scanner.Text(), " ")
  for i := 0; i < n; i++ {
    // t: a temperature expressed as an integer ranging from -273 to 5526
    t, _ := strconv.ParseFloat(inputs[i], 32)
    if math.Abs(t) < math.Abs(minT) || math.Abs(t) == math.Abs(minT) && t > minT {
      minT = t
    }
  }
  
  if n == 0 {
    fmt.Println(0)
  } else {
    fmt.Println(int(minT))
  }
}
